ansible/roles/invoker/tasks/deploy.yml

```diff
@@ -23,9 +23,23 @@
   when: docker_registry != ""
   retries: 3
 
-- name: determine docker root dir
+- name: "determine docker root dir on docker-machine"
   uri:  url="http://{{ inventory_hostname }}:{{ docker.port }}/info" return_content=yes
-  register: dockerInfo
+  register: dockerInfo_output
+  when: "'environments/docker-machine' in inventory_dir"
+
+- set_fact:
+    dockerInfo: "{{ dockerInfo_output['json'] }}"
+  when: "'environments/docker-machine' in inventory_dir"
+
+- name: "determine docker root dir"
+  shell: curl --unix-socket /var/run/docker.sock http:/v1.24/info
+  register: dockerInfo_output
+  when: "'environments/docker-machine' not in inventory_dir"
+
+- set_fact:
+    dockerInfo: "{{ dockerInfo_output.stdout|from_json }}"
+  when: "'environments/docker-machine' not in inventory_dir"
```

Comment:
This feels like a step back and it clutters  environmental differences even more. Can we somehow fold this into one setting? I.e.: docker_host for docker-machine would be: 192.168.99.100:4243 and for docker-on-mac it would be /var/run/docker.sock where you could detect that you try to use a socket or an IP which would at least remove the environment hard-coding. WDYT?